File: 2021/py/day01.py
from aocd import get_data
import pandas as pd


def parse_data():
    data = get_data(year=2021, day=1)
    return pd.DataFrame({"seq": [int(line) for line in data.splitlines()]})


def part_one(df):
    df["inc"] = df["seq"] > df["seq"].shift(1)
    return df["inc"].sum()


def part_two(df):
    df["window"] = df["seq"].rolling(3).sum()
    df["inc"] = df["window"] > df["window"].shift(1)
    return df["inc"].sum()


if __name__ == "__main__":
    df = parse_data()
    print("*" * 10, "Part One", "*" * 10)
    print(part_one(df))
    print("*" * 10, "Part Two", "*" * 10)
    print(part_two(df))

File: 2021/ts/day01.ts
import { readFileSync } from 'fs';

const rawData = readFileSync('data.txt', 'utf-8');

function parseData(rawData: string) {
    let lines = rawData.split('\n');
    let sequence = lines.map(i => parseInt(i));
    return sequence
}

function partOne(sequence: number[]) {
    let count = 0;
    for (let i = 1; i < sequence.length; i++) {
        if (sequence[i] > sequence[i-1]) {
            count += 1;
        }
    }
    return count;
}

function partTwo(sequence: number[]) {
    let count = 0;
    let window: number[] = []
    for (let i = 2; i < sequence.length; i++) {
        let current = sequence[i-2] + sequence[i-1] + sequence[i];
        window.push(current);
    }
    for (let i = 1; i < window.length; i++) {
        if (window[i] > window[i-1]) {
            count += 1;
        }
    }
    return count;
}


let data = parseData(rawData)

console.log("Part One")
let partOneAnswer = partOne(data)
console.log(partOneAnswer)

console.log("Part Two")
let partTwoAnswer = partTwo(data)
console.log(partTwoAnswer)
